Letterbook.Api/Controllers/WebFingerController.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Letterbook.Core;
+using Letterbook.Core.Models.WebFinger;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Letterbook.Api.Controllers;
+
+public class WebFingerController
+{
+    private readonly IAccountService _accountService;
+
+    public WebFingerController(IAccountService accountService)
+    {
+        _accountService = accountService;
+    }
+
+    [HttpGet]
+    [Route("/.well-known/webfinger")]
+    public ActionResult Get([FromQuery]string resource)
+    {
+        return new OkObjectResult(_accountService.LookupAccount(WebFingerQueryTarget.Parse(resource)));
+    }
+}
```

Comment:
It's been a minute since I looked at the webfinger spec, but I think there is some more of it that can be implemented in the API adapter, even without any real lookup. Specifically, shouldn't it check that the domain in the query target is one that we handle? You can get the configured domain from the core options.

Reply:
Likely yep, implementation is for sure incomplete